src/main/java/org/cryptomator/common/CatchingExecutors.java

```diff
@@ -22,12 +23,28 @@ public final class CatchingExecutors {
 
 	private CatchingExecutors() { /* NO-OP */ }
 
-	public static class CatchingScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor {
+	/**
+	 * Executor suitable for scheduled, <em>one-shot</em> Tasks.
+	 * <p>
+	 * This executor does not support repeated execution, see {@link OneShotScheduledExecutorService}
+	 */
+	public static class CatchingScheduledThreadPoolExecutor extends ScheduledThreadPoolExecutor implements OneShotScheduledExecutorService {
 
 		public CatchingScheduledThreadPoolExecutor(int corePoolSize, ThreadFactory threadFactory) {
 			super(corePoolSize, threadFactory);
 		}
 
+		@Override
+		public ScheduledFuture<?> scheduleAtFixedRate(Runnable command, long initalDelay, long period, TimeUnit unit) {
+			throw new UnsupportedOperationException("This is an one-shot executor service!");
```

Comment:
Instead of extending `ScheduledThreadPoolExecutor` but then forbidding methods (that are publicly accessible and won't cause any compile time error), how about encapsulating the command into a one-shot execution. This way the only task that gets scheduled at a fixed rate is this simple one-liner that we can assert never to throw anything.

```suggestion
			var oneShot = () -> this.execute(command);
			return super.scheduleAtFixedRate(oneShot, initialDelay, period, unit);
```